pypeit/pypeit.py

```diff
@@ -11,68 +11,76 @@
 from collections import OrderedDict
 
 from pypeit import msgs
-from pypeit.core import fsort
-from pypeit.core import qa
-from pypeit.par.util import make_pypeit_file, parse_pypeit_file
-
 from pypeit import pypeitsetup
-from pypeit.scripts import run_pypeit
-from pypeit.core import pypsetup
 from pypeit import calibrations
 from pypeit import scienceimage
-from pypeit.core import wave
 from pypeit import specobjs
-from pypeit.core import save
 from pypeit import fluxspec
+from pypeit.core import paths
+from pypeit.core import qa
+from pypeit.core import pypsetup
+from pypeit.core import wave
+from pypeit.core import save
 from pypeit.core import load
+from pypeit.spectrographs.spectrograph import Spectrograph
+from pypeit.scripts import run_pypeit
+from pypeit.par.util import make_pypeit_file, parse_pypeit_file
 
 from pypeit import debugger
 
-
 class PypeIt(object):
     """
     This class is designed to run PypeIt
 
     .. todo::
         Improve docstring...
 
-    Parameters
-    ----------
-    spectrograph: Spectrograph
-    setups_path: str, optional
-      Path for files related to all setups
-    verbosity: int, optional
-      2=verbose
-    overwrite: bool, optional
-      Overwrite files
-    logname: str, optional
-      filename for log file
-    show: bool, optional
-      Show QA on screen along the way
-
-    Attributes
-    ----------
+    Args:
+        spectrograph (:obj:`str`,
+            :class:`pypeit.spectrographs.spectrograph.Spectrograph`):
+            The string or `Spectrograph` instance that sets the
+            instrument used to take the observations.  Used to set
+            :attr:`spectrograph`.
+        setups_path (:obj:`str`, optional):
+            Path for files related to all setups
+        verbosity (:obj:`int`, optional):
+            Verbosity level of system output.  Can be::
+                - 0: No output
+                - 1: Minimal output (default)
+                - 2: All output
+        overwrite (:obj:`bool`, optional):
+            Flag to overwrite any existing files/directories.
+        logname (:obj:`str`, optional):
+            The name of an ascii log file with the details of the
+            reduction.
+        show: (:obj:`bool`, optional):
+            Show reduction steps via plots (which will block further
+            execution until clicked on) and outputs to ginga. Requires
+            remote control ginga session via "ginga --modules=RC &"
+
+    Attributes:
         pypeit_file (:obj:`str`):
             Name of the pypeit file to read.  Pypit files have a specific
             set of valid formats. A description can be found `here`_
             (include doc link).
-
-    Inherited Attributes
-    --------------------
     """
     __metaclass__ = ABCMeta
 
     def __init__(self, spectrograph, setups_path=None, verbosity=2,
                  overwrite=True, logname=None, show=False):
 
+        # Spectrometer class
```

Comment:
Good point.  I just moved these tests into `load_spectrograph`.  I.e., you can now pass `load_spectrograph` a `Spectrograph` instance, that it will simply return, or a string.  I'm not crazy about the idea that `load_spectrograph` can return `None` because it means that the relevant class is not well defined.  I understand why this is allowed for the master frames, but it could lead to problems for users.